tock/hours/views.py

```diff
@@ -800,26 +802,30 @@ def last_timecard(self):
 
     def prefilled_formset(self):
         timecard = self.last_timecard()
-        project_ids, extra = [], 1
+        prefills = dict(TimecardPrefillData.objects.filter(
+            employee=self.request.user.user_data).values_list(
+            'project', 'hours'))
+        project_ids = []
+        extra = 1
         if timecard:
-            project_ids = set(
+            project_ids = [
                 tco.project_id for tco in
                 timecard.timecardobjects.all()
-            )
+            ]
             extra = len(project_ids) + 1
 
         rp = ReportingPeriod.objects \
             .prefetch_related('holiday_prefills__project') \
             .get(start_date=self.kwargs['reporting_period'])
 
-        init = []
         if rp.holiday_prefills:
-            init += [
-                {'hours_spent': hp.hours_per_period, 'project': hp.project.id}
-                for hp in rp.holiday_prefills.all()
-            ]
-        init += [{'hours_spent': None, 'project': pid} for pid in project_ids]
+            for hp in rp.holiday_prefills.all():
```

Comment:
A comment might be good here. It starts to get a bit unclear what's going on

Reply:
Duly noted, thanks!